pyro/optim/optim.py

```diff
@@ -40,14 +43,19 @@ def __call__(self, params,  *args, **kwargs):
         Do an optimization step for each param in params. If a given param has never been seen before,
         initialize an optimizer for it.
         """
-
+        if hasattr(self, 'pt_scheduler_constructor'):
```

Comment:
nit: `hasattr` smells bad; does it make sense to remove these two lines and instead
```py
class PyroLRScheduler(PyroOptim):
    def __call__(self, params,  *args, **kwargs):
        kwargs['epoch'] = self.epoch
        super(PyroLRScheduler, self).__call__(params, *args, **kwargs)
```